content/learn/book/migration-guides/0.7-0.8/_index.md

```diff
@@ -0,0 +1,401 @@
+
+Replace calls to `EntityMut::get_unchecked` with calls to `EntityMut::get`.
+
+### [Replace ReadOnlyFetch with ReadOnlyWorldQuery](https://github.com/bevyengine/bevy/pull/4626)
+
+The trait `ReadOnlyFetch` has been replaced with `ReadOnlyWorldQuery` along with the `WorldQueryGats::ReadOnlyFetch` assoc type which has been replaced with `<WorldQuery::ReadOnly as WorldQueryGats>::Fetch`
+
+The trait `ReadOnlyFetch` has been replaced with `ReadOnlyWorldQuery` along with the `WorldQueryGats::ReadOnlyFetch` assoc type which has been replaced with `<WorldQuery::ReadOnly as WorldQueryGats>::Fetch`
+
+* Any where clauses such as `QueryFetch<Q>: ReadOnlyFetch` should be replaced with `Q: ReadOnlyWorldQuery`.
+* Any custom world query impls should implement `ReadOnlyWorldQuery` insead of `ReadOnlyFetch`
+
+Functions `update_component_access` and `update_archetype_component_access` have been moved from the `FetchState` trait to `WorldQuery`
+
+* Any callers should now call `Q::update_component_access(state` instead of `state.update_component_access` (and `update_archetype_component_access` respectively)
+* Any custom world query impls should move the functions from the `FetchState` impl to `WorldQuery` impl
+
+`WorldQuery` has been made an `unsafe trait`, `FetchState` has been made a safe `trait`. (I think this is how it should have always been, but regardless this is _definitely_ necessary now that the two functions have been moved to `WorldQuery`)
+
+* If you have a custom `FetchState` impl make it a normal `impl` instead of `unsafe impl`
+* If you have a custom `WorldQuery` impl make it an `unsafe impl`, if your code was sound before it is going to still be sound
+
+### [Fix unsoundness with Or/AnyOf/Option component access'](https://github.com/bevyengine/bevy/pull/4659)
+
+Query conflicts from `Or`/`AnyOf`/`Option` have been fixed, and made stricter to avoid undefined behaviour.
+If you have new query conflicts due to this you must refactor your systems; consider using `ParamSet`.
+
+### [Remove task_pool parameter from par_for_each(_mut)](https://github.com/bevyengine/bevy/pull/4705)
+
+The `task_pool` parameter for `Query(State)::par_for_each(_mut)` has been removed. Remove these parameters from all calls to these functions.
+
+Before:
+
+```rust
+fn parallel_system(
+   task_pool: Res<ComputeTaskPool>,
+   query: Query<&MyComponent>,
+) {
+   query.par_for_each(&task_pool, 32, |comp| {
+        ...
+   });
+}
+```
+
+After:
+
+```rust
+fn parallel_system(query: Query<&MyComponent>) {
+   query.par_for_each(32, |comp| {
+        ...
+   });
+}
+```
+
+If using `Query` or `QueryState` outside of a system run by the scheduler, you may need to manually configure and initialize a `ComputeTaskPool` as a resource in the `World`.
+
+### [Fail to compile on 16-bit platforms](https://github.com/bevyengine/bevy/pull/4736)
+
+`bevy_ecs` will now explicitly fail to compile on 16-bit platforms, as the behavior was undefined.
```

Comment:
I don't think there was any unconditional UB - only unsoundness?

Reply:
@alice-i-cecile that was your suggestion to change this. Do you have a preference for this? Otherwise I can just reuse the original test from the PR that didn't mention UB.
```suggestion
`bevy_ecs` will now explicitly fail to compile on 16-bit platforms, , due to an inability to write sound unsafe code for these platforms.
```

Reply:
Yeah, I'm fine to accept that suggestion.